library/ecdsa.c

```diff
@@ -289,22 +289,22 @@ int mbedtls_ecdsa_verify( mbedtls_ecp_group *grp,
 /*
  * Convert a signature (given by context) to ASN.1
  */
-static int ecdsa_signature_to_asn1( const mbedtls_mpi *r, const mbedtls_mpi *s,
-                                    unsigned char *sig, size_t *slen )
+int mbedtls_ecdsa_signature_to_asn1( const mbedtls_mpi *r, const mbedtls_mpi *s,
+                             unsigned char *sig, size_t *slen, size_t ssize )
 {
```

Comment:
If this function fails (because `MBEDTLS_ASN1_CHK_ADD` returns an error), it shouldn't leave half-written output in `sig`. (Even before, it shouldn't have left some half-written data on the stack, but it's worse in an output parameter.) Since `MBEDTLS_ASN1_CHK_ADD` contains `return`, the easiest way to resolve this would be a wrapper function
```
{
    int ret = internal_ecdsa_signature_to_asn1( r, s, sig, slen, ssize );
    if( ret != 0 )
        memset( sig, ssize, 0 );
    return( ret );
}
```

Reply:
Done